Assets/_Laz/Scripts/Managers/WorldManager.cs

```diff
@@ -61,29 +61,34 @@ public void Reset()
                 patrolItem.Reset();
             }
         }
-        
+
         private void Awake()
         {
             var interests = GameObject.FindGameObjectsWithTag(Tags.LazoInterest);
             var objectsOfInterest = GenerateObjectOfInterest(interests);
             var laz = new LazPlayer();
 
             StateManagerInstance.OnStateChanged += HandleOnStateChanged;
-            
+
+            if (_checkPointManager != null)
+            {
+                _checkPointManager.Initialize();
```

Comment:
after some thought, I think you can pass in `laz` into this Initialize(). 

So when there's a new check point, you have the Laz object already to Set the Spawn Location